docs/application/dotnet/guides/nui/grid-layout.md

```diff
@@ -1,29 +1,110 @@
+
+ - On horizontal orientation, the children will automatically occupy empty cells from left to right.
+ - On vertical orientation, the children will automatically occupy empty cells from top to bottom.
+
+The following figure shows how the `GridOrientation` property works:
+
+![gridOrientation](./media/gridOrientation.png)
+
+## Column spacing and row spacing
+
+The `ColumnSpacing` and `rowSpacing` properties are the distance between columns or rows. In order to set same space between the children, you can use `ColumnSpacing` and `rowSpacing` properties instead of setting the same margin for each child.
+
+The following figure shows how the `ColumnSpacing` and `RowSpacing` properties work:
+
+![gridSpacing](./media/gridSpacing.png)
+
+The following table describes the properties that affect the specified child:
+
+| Property               | Type            | Description |
+| -----------------------| --------------- | ----------- |
+| `Column`               | int             | Gets or sets the column index of the child. |
+| `Row`                  | int             | Gets or sets the row index of the child. |
+| `ColumnSpan`           | int             | Gets or sets the number of columns that the child occupy. |
+| `RowSpan`              | int             | Gets or sets the number of rows that the child occupy. |
+| `HorizontalStretch`    | StretchFlags    | Gets or sets the value how child is resized within horizontal space of the cell where child occupy. |
+| `VerticalStretch`      | StretchFlags    | Gets or sets the value how child is resized within vertical space of the cell where child occupy. |
+| `HorizontalAlignment`  | Alignment       | Gets or sets the horizontal alignment of the child within the cell. |
+| `VerticalAlignment`    | Alignment       | Gets or sets the vertical alignment of the child within the cell. |
+
+## Column and row
+
+The `Column` and `row` properties specify the cell index of the child. If these properties aren't assigned, they are automatically set depending on [`GridOrientation`](#grid-orientation), [`Columns`](#columns-and-rows) and [`Rows`](#columns-and-rows).
+
+The following figure shows how to position child on the grid using `Column` and `Row` properties:
+
+![gridColumnRow](./media/gridColumnRow.png)
+
+## Column span and row span
+
+The `ColumnSpan` and `rowSpan` properties specify the number of cells that the child can occupy. The sum of cell index and cell span should be less than the maximum number of cells specified by `Columns` or `Rows` properties.
+
+The following figure shows how to position child on the grid using `Column` and `ColumnSpan` properties:
+
+![gridColumn](./media/gridColumn.png)
+
+## Horizontal stretch and vertical stretch
+
+The `HorizontalStretch` and `VerticalStretch` properties define how a child is resized within the space of the cell. The values can be specified as `Stretchflags` enum:
```

Comment:
Since these are properties for child of `GridLayout`, you need to use `SetHorizontalStretch()` or `SetVerticalStretch()` in `.cs`.
`GridLayout.HorizontalStretch` and `GridLayout.VerticalStretch` are available on xaml. for example,
```xaml
<View GridLayout.HorizontalStretch="ExpandAndFill" GridLayout.VerticalStretch="ExpandAndFill"/>
```

Reply:
@dasg34, thanks for clarifying, could you also answer the other ones?